cmd/synthetic-monitoring-agent/main.go

```diff
@@ -133,6 +133,12 @@ func run(args []string, stdout io.Writer) error {
 		}
 	}
 
+	if !config.DisableK6 {
+		if err := features.Set(feature.K6); err != nil {
+			return fmt.Errorf("cannot set k6 feature: %w", err)
+		}
+	}
```

Comment:
It's not fully clear to me how the new flag (`--disable-k6`) and the old one (`--features=k6`) interact.

I think it makes sense having a dedicated K6 flag for disabling k6, but that makes me wonder if we should get rid of, or deprecate the feature flag.

Reply:
This change does get rid of the `--features` flag on L105-106. I could "formally" deprecate it by detecting it and printing out a message.

---

**How the new vs old flags interact**

The old flag `--features` is captured by `flags.Var`, which passes the cmd argument to the `Set` method of the container object. `features.Set` assumes a csv string and splits it to save each element.

Example: `features.Set("traceroute,k6,adhoc")` -> `map[string]struct{}{"traceroute": struct{}, "k6": struct{}, "adhoc": struct{}}`.

With this change, if `--disable-k6` is `false` or unset, we explicitly call `features.Set("k6")` to mark k6 as enabled by default.

Reply:
Ah, you're definitely right, I did not see the removal of the `--features` CLI flag 🙈 
Looking good :)